packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1105,10 +1105,162 @@ class EditableText extends StatefulWidget {
   /// The minimum platform SDK version that supports Autofill is API level 26
   /// for Android, and iOS 10.0 for iOS.
   ///
+  /// ### iOS-specific Concerns:
+  ///
+  /// To provide the best user experience and ensure your app fully supports
+  /// password autofill on iOS, follow these steps:
+  ///
+  /// - Set up your iOS app's
+  ///   [associated domains](https://developer.apple.com/documentation/safariservices/supporting_associated_domains_in_your_app).
+  /// - Some autofill hints only work with specific [keyboardType]s. For example,
+  ///   [AutofillHints.name] requires [TextInputType.name] and [AutofillHints.email]
+  ///   works only with [TextInputType.email]. Make sure the input field has a
+  ///   compatible [keyboardType].
```

Comment:
I think the 2nd bullet point needs better documentation on which keyboard types and autofill hints are compatible. I did some tests on an iOS 13.5 device using a native app, most `UITextContentType` only works with `UIKeyboardType.namePhonePad` (which is mapped to `TextInputType.name` in this PR), including content types like `.streetAddress`, `.zipCode` and `.telephoneNumber` (despite there're more fitting keyboard types). I can't really make sense of it so I'm not sure we should put that behavior in public documentation.

Reply:
Maybe our users should be seeking out native iOS documentation on this stuff anyway, and we should just be clear about how we map to native.

Reply:
Unfortunately the behavior isn't documented there either. Maybe I should change the name `autofillHints` to `contentType` to emphasize that it's not only for autofill, and it doesn't guarantee the platform autofill will trigger?